tests/serverpod_test_server/test_integration/database_operations/transactions/transaction_or_rollback_test.dart

```diff
@@ -0,0 +1,581 @@
+            expect(identical(transaction1, transaction2), isTrue);
+
+            await SimpleData.db.insertRow(
+              session,
+              SimpleData(num: 2),
+              transaction: transaction2,
+            );
+
+            expect(
+              await SimpleData.db.count(session, transaction: transaction1),
+              2,
+            );
+            expect(
+              await SimpleData.db.count(session, transaction: transaction2),
+              2,
+            );
+          }, transaction: transaction1);
+
+          expect(
+            await SimpleData.db.count(session, transaction: transaction1),
+            2,
+          );
+        }, transaction: null);
+
+        expect(
+          await SimpleData.db.count(session),
+          2,
+        );
+      });
+
+      test(
+          'when throwing in a nested savepoint callback, '
+          'then the parent transaction can still be completed and its data be visible on the outside.',
+          () async {
+        await session.transactionOrSavepoint((final transaction1) async {
+          await SimpleData.db.insertRow(
+            session,
+            SimpleData(num: 1),
+            transaction: transaction1,
+          );
+
+          try {
+            await session.transactionOrSavepoint((final transaction2) async {
+              await SimpleData.db.insertRow(
+                session,
+                SimpleData(num: 2),
+                transaction: transaction1,
+              );
+
+              expect(
+                await SimpleData.db.count(session, transaction: transaction2),
+                2,
+              );
+
+              throw _ForcedTestException();
+            }, transaction: transaction1);
+          } on _ForcedTestException catch (_) {}
+        }, transaction: null);
+
+        expect(
+          await SimpleData.db.count(session),
+          1,
+        );
+      });
+
+      test(
+          'when throwing in a nested transaction callback, '
+          'then the the data written with the transaction is not visible on the outside.',
+          () async {
+        try {
+          await session.transactionOrSavepoint(
+            (final transaction) async {
+              await SimpleData.db.insertRow(
+                session,
+                SimpleData(num: 1),
+                transaction: transaction,
+              );
+              await SimpleData.db.insertRow(
+                session,
+                SimpleData(num: 2),
+              );
+
+              expect(
+                await SimpleData.db.count(session, transaction: transaction),
+                2,
+              );
+
+              throw _ForcedTestException();
+            },
+            transaction: null,
+          );
+        } on _ForcedTestException catch (_) {}
+
+        expect(
+          await SimpleData.db.count(session),
+          1,
+        );
+      });
+    },
+    rollbackDatabase: RollbackDatabase.disabled,
```

Comment:
Use [TestTags.concurrencyOneTestTag],


as in: tests/serverpod_test_server/test_integration/database_operations/transactions/transaction_isolatation_level_test.dart

Reply:
Added the tag as suggested, @SandPod. Interestingly so far CI was already passing all the time. Likely all the others that would've interfered excluded themselves nicely so far, that one additional test file didn't make a difference. But then this is solid going forward :)